### ISSUE TITLE
Enhancement: Symmetric spaced comment

### ISSUE THREAD

**Issue by andy-hanson** [eslint/eslint#4133]
I've noticed that this code:

	/* spaced-comment*/

doesn't trigger a warning.

Here's my package.json config (eslint 1.6.0):

	{
		"eslintConfig": {
			"rules": {
				"spaced-comment": 1
			}
		}
	}

This isn't a bug; the rule does say "Requires or disallows a whitespace (space or tab) beginning a comment".
I want to check for spaces on both sides of the comment.

Could there be an option to make that code trigger a warning `Expected space or tab before */ in comment`,
so that only the code below would pass?

	/* spaced-comment */

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
Presumably that only applies to block comments, right? Line comments end on
a newline anyway.

**Comment by andy-hanson** [comment]
Yes, it should only apply to block comments.

**Comment by platinumazure** [comment]
For what it's worth, I like it. What happens for multiple-line block comments, though? Should this be single line block comments only?

```js
/* Could be a problem with comments like this?
*/

**Comment by andy-hanson** [comment]
Presumably you were expecting code to look like this:

	/* Could be a problem
	 * with comments
	 * like this?
	 */

And not like this:

	/* Could be a problem
	 * with comments
	 * like this?
	*/

A missing space at the end is only a problem if you're trying to align stars, which sounds like a different rule entirely.
Personally I prefer to write multi-line comments like this:

	/*
	multi-line
	comment
	*/

which I don't think should trigger a warning. Although this should:

	/* multi-line
	comment*/

The rule being that `*/` should be preceded by some whitespace, be it a space or a newline.

**Comment by gyandeeps** [comment]
@eslint/eslint-team Thoughts?

**Comment by IanVS** [comment]
I like the concept.  Would this be an option to the existing `spaced-comment` rule?  If so, what should it be called?

**Comment by nzakas** [comment]
Yup, it would be a new option. Looking for naming ideas.

**Comment by nzakas** [comment]
@eslint/eslint-team does anyone want to champion this?

**Comment by BYK** [comment]
I'm willing to work on this.

**Comment by nzakas** [comment]
And so it shall be. Please be sure to propose some option names here before getting started.

**Comment by kaicataldo** [comment]
Late to the party, but I'm :+1: for this too!

**Comment by BYK** [comment]
Proposal:

`alignSpacing: true` and `alignMarkers: ['*']`

**Comment by btmills** [comment]
@BYK those options seem like they would be for vertically aligning block comments, and I agree with @andy-hanson that that sounds like a separate rule. Perhaps I'm misunderstanding? What about `symmetric: true`?

**Comment by BYK** [comment]
@btmills - `alignBlock: true` + `symmetricBlock: true`?

**Comment by BYK** [comment]
Or, `block: { align: true, symmetric: true, markers: ['*'] }`

**Comment by btmills** [comment]
@BYK so you do want to do vertical alignment as part of this as well? If so I think `block: { align: true, symmetric: true, markers: ['*'] }` works. How would you configure it for the following?

```js
/*
multi-line
comment
*/
```

**Comment by nzakas** [comment]
I think `symmetric` is a difficult option name for non-native English speakers. Maybe `mirror` instead?

**Comment by BYK** [comment]
@btmills - Yup, I want that. That said the rule will become more like `comment-style` at that point since vertical alignment has nothing to do with spacing so I give you that. For the sake of completeness though, I'd use `block: { align: true, symmetric: whatever, markers: [] } ` for that I guess.

@nzakas - I'm not so sure. Symmetric is quite well-known in Turkish but I cannot comment for other languages. Also what do you think about @btmills' concern. I think he has a point. We can just address symmetric in this issue and create a new rule, `aligned-comment` for others. That new rule can even cover single line comment aligning when there are consequent single line comments.

**Comment by pedrottimark** [comment]
While you are thinking about wording, we just replaced “vertical alignment” with “horizontal alignment of adjacent X” (in this case I think X = lines) in key-spacing. What do y’all think?

My mind is still so full of CSS that it makes me think of `vertical-align: baseline` 🤓

**Comment by nzakas** [comment]
@byk not everyone speaks Turkish. :) We try to keep options using simple words so they are more easily understandable to everyone. 

Overall, I think we have expanded the scope of this issue too far. The original request was just to mirror spacing preferences at the end of an inline comment. I think anything else should be a separate issue (or issues).

**Comment by BYK** [comment]
@nzakas hah :) Anyways, agree with the scope. Still voting for `symmetric` though. @nzakas are you gonna be the tie breaker on that or shall we get more thoughts from @eslint/eslint-team?

**Comment by nzakas** [comment]
@BYK no way I'm going to yield to `symmetric` when `mirror` means the same thing and is more approachable. For reference (via http://www.wordcount.org/main.php):

* "mirror" is the 2,560th most used English word
* "symmetric" is the 14,070th most used English word

Also open to other suggestions, but no way "symmetric" is appropriate for this.

**Comment by BYK** [comment]
The reason I insisted on `symmetric` is although it is a less frequently used word, it conveys the meaning better to me. I wouldn't understand what `mirror` meant in this context but a "symmetric space" makes a lot of space. That said I may be an exception.

Anyways, I'll start implementing the changes. In the meantime we can discuss the name. But I'm not really a fan of `mirror`.

**Comment by alberto** [comment]
Here is another suggestion: `balanced`

**Comment by BYK** [comment]
👍 to `balanced`

**Comment by nzakas** [comment]
Sounds good to me.

**Comment by azhang496** [comment]
hey @BYK are you still working on this? if not i'd like to take a shot at it

**Comment by mikesherov** [comment]
@azhang496 I also have 1 or 2 issues assigned to me if you'd like to take anything. Just let me know. Thanks for contributing!

**Comment by BYK** [comment]
@mikesherov I totally am not, thanks for asking! Removed my face.

**Comment by mikesherov** [comment]
@azhang496, it's all you!

**Comment by azhang496** [comment]
awesome. @mikesherov totally down to take on those issues assigned to you as well!

**Comment by nzakas** [comment]
@azhang496 thanks so much! If you want any realtime help as you're going along, stop by https://gitter.im/eslint/eslint and say hi. :)

**Comment by azhang496** [comment]
just made a PR for this issue - https://github.com/eslint/eslint/pull/6575
this is my first PR so i'd appreciate any & all feedback! :)
